bindings/go/plugin/manager/registries/blobs/blobs.go

```diff
@@ -10,10 +10,25 @@ import (
 
 // CreateBlobData creates a blob based on the location.
 func CreateBlobData(location types.Location) (blob.Blob, error) {
+	var b blob.Blob
+	var err error
+
 	switch location.LocationType {
 	case types.LocationTypeLocalFile:
-		return filesystem.GetBlobFromOSPath(location.Value)
+		b, err = filesystem.GetBlobFromOSPath(location.Value)
 	default:
 		return nil, fmt.Errorf("unsupported location type: %s", location.LocationType)
 	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if location.MediaType != "" {
+		if mtOverrideable, ok := b.(blob.MediaTypeOverrideable); ok {
+			mtOverrideable.SetMediaType(location.MediaType)
+		}
+	}
+
+	return b, nil
```

Comment:
This will be replaced with 
```go
// CreateBlobData creates a blob based on the location.
func CreateBlobData(location types.Location) (blob.ReadOnlyBlob, error) {
	var b blob.Blob
	var err error

	switch location.LocationType {
	case types.LocationTypeLocalFile:
		b, err = filesystem.GetBlobFromOSPath(location.Value)
	default:
		return nil, fmt.Errorf("unsupported location type: %s", location.LocationType)
	}
	if err != nil {
		return nil, err
	}

	w := direct.New(direct.NewReadCloserWrapper(b), direct.WithMediaType(location.MediaType))
	return w, nil
}
```

Once https://github.com/open-component-model/open-component-model/pull/593 is merged.